dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java

```diff
@@ -12,45 +13,63 @@
 import com.liferay.portal.language.LanguageException;
 import com.liferay.portal.language.LanguageUtil;
 import com.liferay.portal.model.User;
-
-import org.apache.velocity.context.Context;
+import com.liferay.util.StringPool;
 import org.apache.velocity.tools.view.context.ViewContext;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-
 import javax.servlet.http.HttpServletRequest;
-
+import java.util.*;
+
+/**
+ * Implementation class for the {@link LanguageAPI}.
+ * 
+ * @author root
+ * @version N/A
+ * @since Mar 22, 2012
+ *
+ */
 public class LanguageAPIImpl implements LanguageAPI {
 
-	private HttpServletRequest request;
-	Context ctx;
 
-	public void init(Object obj) {
-		ViewContext context = (ViewContext) obj;
-		this.request = context.getRequest();
-		ctx = context.getVelocityContext();
-	}
+    private final static LanguageKeyComparator LANGUAGE_KEY_COMPARATOR = new LanguageKeyComparator();
 
+	private HttpServletRequest request; // todo: this should be decouple from the api
 	private LanguageFactory factory;
+	private LanguageVariableAPI languageVariableAPI;
+	
+	/**
+	 * Inits the service with the user {@link ViewContext}
+	 * @param obj
+	 */
+	// TODO: remove me
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)